internal/p2p/peer_test.go

```diff
@@ -231,4 +232,8 @@ func (rp *remotePeer) nodeInfo() types.NodeInfo {
 		Channels:        rp.channels,
 		Moniker:         "remote_peer",
 	}
+	if rp.Network != "" {
```

Comment:
How is the remote peer's network initially set?

Reply:
Ah, good question. So it's set above on [line 230](https://github.com/tendermint/tendermint/pull/6630/files#diff-e69199c774ff088e01b74263b9be6b0f886b17839d4b9cf201f277b1e55de2ddR230). This basically acts as a default `Network`. The `remotePeer` type was modified to include a `Network` field in this pull request. This way, someone constructing a `remotePeer` can override this default if they need to. This was needed for this test so that the peer could be setup as on a different network from the node. An example of this can be seen in [switch_test.go](https://github.com/tendermint/tendermint/pull/6630/files#diff-f42b355038dcf915d1b4faa4377782f1db0a78deb818c39aa2f9fa3318e000d8R222).

Reply:
Ohhh I didn't realize this was a test 😆 

So I presume we _already_ set the `network` field in the non-test type?

Reply:
Ah, yup! In the non-test code, this information is created in the [handshake](https://github.com/tendermint/tendermint/blob/69cf90c51e09a09de6d5c726db1b71842b0c647b/internal/p2p/transport_mconn.go#L333) code.